users/userRouter.js

```diff
@@ -1,47 +1,163 @@
 const express = require('express');
+const users = require('../users/userDb');
+const posts = require('../posts/postRouter');
 
 const router = express.Router();
 
-router.post('/', (req, res) => {
-  // do your magic!
+router.post('/', validateUser(), async (req, res) => {
+  const { newUser } = req.body.name;
+  users.insert({ newUser })
+  .then(newUser => {
+    res.status(201).json(newUser)
+  })
+  .catch(err => {
+    res.status(500).json({
+      errorMessage: "Error saving user to the database."
+    })
+  });
 });
 
-router.post('/:id/posts', (req, res) => {
-  // do your magic!
+router.post('/:id/posts', validateUserId(), validatePost(), (req, res) => {
```

Comment:
validatePost should come before validateUserId because validatePost does not require DB resources